src/Tempest/Database/QueryStatement.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database;
+
+use RuntimeException;
+use Stringable;
+use Tempest\Database\Drivers\MySqlDriver;
+use Tempest\Database\Drivers\PostgreSqlDriver;
+use Tempest\Database\Drivers\SQLiteDriver;
+use UnhandledMatchError;
+
+final class QueryStatement implements Stringable
+{
+    private array                   $query = [];
+
+    public function __construct(
+        private readonly DatabaseDriver $driver,
+        private string $table,
+    ) {
+    }
+
+    public function create(callable $callback): self
+    {
+        if (! empty($this->query)) {
+            throw new RuntimeException('create statement should be the first statement');
+        }
+
+        $this->query[] = sprintf(
+            "CREATE TABLE %s (%s);",
+            $this->table,
+            $callback($this->driver->createQueryStatement($this->table))
+        );
+
+        return $this;
+    }
+
+    /** @throws UnhandledMatchError */
+    public function alterTable(string $action, callable $callback): self
+    {
+        if (! empty($this->query)) {
+            throw new RuntimeException('alter statement should be the first statement');
+        }
+
+        $operation = match ($this->driver::class) {
```

Comment:
I think we might refactor at a later point to have dedicated grammar classes, though that's not necessary for now.

Reply:
I like that idea